extension-manifest-v2/src/classes/BugDb.js

```diff
@@ -10,246 +10,135 @@
-	 */
-	processList(fromMemory, bugs, skip_cache_flush) {
-		// deep cloning bugs all at once is too slow
-		const { patterns } = bugs;
-		const regexes = patterns.regex;
-		const db = {
-			apps: bugs.apps,
-			bugs: bugs.bugs,
-			firstPartyExceptions: bugs.firstPartyExceptions,
-			patterns: {
-				host: patterns.host,
-				host_path: patterns.host_path,
-				path: patterns.path,
-				// regexes are initialized below
-				regex: {}
+		const categoriesMeta = categories.reduce((all, category) => ({
+			...all,
+			[category]: {
+				trackers: [],
+				blockedTrackersCount: 0,
 			},
-			version: bugs.version,
-			JUST_UPDATED_WITH_NEW_TRACKERS: false
-		};
-
-		log('initializing bugdb regexes...');
+		}), {});
 
-		const regexesKeys = Object.keys(regexes);
-		regexesKeys.forEach((id) => {
-			db.patterns.regex[id] = new RegExp(regexes[id], 'i');
-		});
+		patterns.forEach((pattern) => {
+			const id = getPatternId(pattern);
 
-		log('setting bugdb noneSelected/allSelected...');
+			categoriesMeta[pattern.category].trackers.push({
+				id,
+				name: pattern.name,
+				description: '',
+				blocked: selectedApps.hasOwnProperty(id),
+				shouldShow: true,
+				catId: pattern.category,
+				trackerID: pattern.key,
+			});
 
-		const num_selected = size(conf.selected_app_ids);
-		db.noneSelected = (num_selected === 0);
+			if (selectedApps.hasOwnProperty(id)) {
+				categoriesMeta[pattern.category].blockedTrackersCount += 1;
+			}
 
-		// since allSelected is slow to eval, make it lazy
-		defineLazyProperty(db, 'allSelected', () => {
-			const num_selected_lazy = size(conf.selected_app_ids);
-			return (!!num_selected_lazy && every(db.apps, (app, app_id) => conf.selected_app_ids.hasOwnProperty(app_id)));
+			this.db.apps[id] = {
+				name: pattern.name,
+				trackerID: pattern.key,
+				get wtm() {
+					return this.trackerID;
+				},
+				cat: pattern.category,
+			};
+
+			this.db.bugs[id] = {
+				aid: id,
+			};
 		});
 
-		log('processed bugdb...');
-
-		if (!fromMemory) {
-			// if there is an older bugs object in storage update newAppIds and apply block-by-default
-			const old_bugs = conf.bugs;
-			if (old_bugs && old_bugs.hasOwnProperty('version') && (old_bugs.version !== bugs.version)) {
-				const new_app_ids = BugDb.updateNewAppIds(bugs.apps, old_bugs.apps);
-				if (new_app_ids.length > 0) {
-					log(`${new_app_ids.length} new trackers have been added`);
-					const { selected_app_ids } = conf;
-					const newBlockedTrackers = BugDb._scanForNewTrackersToBlock(bugs, old_bugs, new_app_ids, selected_app_ids);
-
-					if (newBlockedTrackers.length > 0) {
-						newBlockedTrackers.forEach((app_id) => {
-							selected_app_ids[app_id] = 1;
-						});
-
-						// (triggers a persist)
-						conf.selected_app_ids = selected_app_ids;
-					}
-					db.JUST_UPDATED_WITH_NEW_TRACKERS = true;
+		/* block new trackers if in mostly blocked category */
+		if (just_upgraded) {
+			Object.values(categoriesMeta).forEach((meta) => {
```

Comment:
Shouldn't we preserve the existing logic (_scanForNewTrackersToBlock) to take into account which categories are enabled/disabled by default?

A use case is when a new category is added, or if the extension for some reason started with an empty set of trackers in one category. As I understand the new code, the new trackers would be disabled now.